corehq/apps/es/case_search.py

```diff
@@ -141,21 +141,25 @@ def case_property_filter(case_property_name, value):
     )
 
 
-def case_property_query(case_property_name, value, fuzzy=False):
+def case_property_query(case_property_name, value, fuzzy=False, multivalue_mode=None):
```

Comment:
Hey Joel, this new functionality is tested by extension in `test_filter_dsl.py` with what I've got in [this commit](https://github.com/dimagi/commcare-hq/pull/31187/commits/7dc27805ffd20a7f28899179cd3418c356163d91). The inputted `selected` functions use `case_property_query` with a `multivalue_mode` and check the ES output.

Are you suggesting I add more specific test cases for just this method, or that these tests should be in `test_case_search_es.py` instead?

Reply:
> Are you suggesting I add more specific test cases for just this method

Yes, in this situation there should be additional tests in `corehq/apps/es/tests/test_case_search_es.py` to test the keyword arg changes made on the `case_property_query()` and `case_property_text_query()` functions. Your `filter_dsl` tests are fine as is (those tests verify logic that's more specific than just these two functions on their own).

I haven't had a look at the existing tests in `corehq/apps/es/tests/test_case_search_es.py`, but the fact that you were able to modify the behavior of these two functions without causing failures suggests that there probably isn't sufficient testing of these functions yet. If that's the true then it would be nice if you added tests that at least verify the functionality provided by your new keyword arguments (bonus points for adding tests for untested functionality you didn't change :grin:).